src/main/scala/io/iohk/ethereum/blockchain/sync/FastSyncBlockHeadersRequestHandler.scala

```diff
@@ -22,6 +24,9 @@ class FastSyncBlockHeadersRequestHandler(
 
     (blockHashes zip blockHeaders.headers).foreach { case (hash, header) =>
       blockHeadersStorage.put(hash, header)
+      val parentTotalDifficulty = totalDifficultyStorage.get(header.parentHash)
+        .getOrElse(throw new Exception(s"Block ${Hex.toHexString(header.parentHash.toArray)} total difficulty not found"))
```

Comment:
You were right, I had forgotten to properly handle it. In my last commit I changed this to handle properly not finding the total difficulty of the parent of a block.